### PR TITLE
Removing some null checks, where we already know that the variable in…

### PR DESCRIPTION
**Pull request by coheigea**
… question is non-null

**Please** add a meaningful description for your change here

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java [hunk 1]
```diff
@@ -49,9 +49,7 @@ public void onMatch(RelOptRuleCall call) {
     final Aggregate aggregate = call.rel(0);
     final Project project = call.rel(1);
     RelNode x = updateWindow(call, aggregate, project);
-    if (x != null) {
-      call.transformTo(x);
-    }
+    call.transformTo(x);
```
Comment (robertwb) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java]: How do we know this x is not null here?

Comment (coheigea) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java]: As "updateWindow" returns "return new BeamAggregationRel(..."?

**Comment by aromanenko-dev** [comment]
Thanks! LGTM

**Comment by coheigea** [comment]
Thanks @aromanenko-dev !

**Pull request merged**